types/index.d.ts

```diff
@@ -16,22 +17,47 @@ declare module 'socketio-jwt' {
   }
 
   /**
-   * Callback gets called, if secret is given dynamically.
-   */
+   * Callback gets called, if secret (success param) is given dynamically.
+   * secret will be passed as 'secretOrPublicKey'
+   * jsonwebtoken.verify(
+   *  token: string,
+   *  secretOrPublicKey: Secret | GetPublicKeyOrSecret,
+   *  options?: VerifyOptions,  callback?: VerifyCallback
+   * )
+  **/
   interface ISocketCallback {
-    (err: ISocketIOError, success: string): void;
+    (err: ISocketIOError, success?: Secret | GetPublicKeyOrSecret): void;
```

Comment:
I get a TypeScript error when I try to pass null for the `err` parameter in the callback. Would suggest also fixing this:
```suggestion
    (err: ISocketIOError | null, success?: Secret | GetPublicKeyOrSecret): void;
```